### ISSUE TITLE
Query result error

### ISSUE THREAD

**Issue by wanpengfei-git** [StarRocks/starrocks#3024]
### Steps to reproduce the behavior (Required)
1. DROP DATABASE IF EXISTS n2;
CREATE DATABASE n2;
USE n2;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE IF NOT EXISTS t0 (c_0_0 BOOLEAN NULL ) AGGREGATE KEY (c_0_0) DISTRIBUTED BY HASH (c_0_0) properties("replication_num"="3") ;
CREATE TABLE IF NOT EXISTS t1 (c_1_0 DECIMAL(34, 13)  NULL ,c_1_1 CHAR(21) NULL ,c_1_2 VARCHAR(11) NULL ,c_1_3 DATETIME NULL ,c_1_4 BOOLEAN NULL ,c_1_5 DECIMAL(26, 2)  NULL ,c_1_6 CHAR(11) NULL ,c_1_7 BOOLEAN NULL ,c_1_8 BOOLEAN NULL ,c_1_9 BOOLEAN NULL ,c_1_10 CHAR(11) NOT NULL ,c_1_11 SMALLINT NOT NULL ,c_1_12 CHAR(21) NULL ,c_1_13 DATE NULL ,c_1_14 BOOLEAN NULL ,c_1_15 BOOLEAN NOT NULL ,c_1_16 DECIMAL(25, 6)  NULL ) DUPLICATE KEY (c_1_0,c_1_1,c_1_2,c_1_3,c_1_4,c_1_5,c_1_6,c_1_7,c_1_8,c_1_9) DISTRIBUTED BY HASH (c_1_4,c_1_5,c_1_6,c_1_7,c_1_9,c_1_0,c_1_2,c_1_8,c_1_1) properties("replication_num"="3") ;
INSERT INTO t1 VALUES (73274266.2559530799767, "x", "y", "1970-01-01 08:00:00", true, 9795931244396562931345.34, "", true, true, true, "Vty", 0, "yyf", "1970-01-01", NULL, false, 7148382338283.990529), (2087436.8227650056287, "oLN", "oWCnikc", "1970-01-01 08:00:00", false, 6013717296855354277422.32, NULL, false, true, NULL, "EJDV8M", 7, "L632W", "1970-01-01", false, false, 3486248642558.852308), (29392878.4859718057343, "Q", "U", "1970-01-01 08:00:00", false, 4330076166445984142754.25, "kVuD", true, false, NULL, "6m", 2, "8pncW", "1970-01-01", true, false, 9034030916002.091349);
INSERT INTO t0 VALUES (false), (true);
INSERT INTO t1 VALUES (NULL, "70", "l", "1970-01-01 08:00:00", false, NULL, "", false, false, false, "4SQ22D", 15, "l", "1970-01-01", false, true, 7019425131308.978094), (15657298.2318305431585, "", "", NULL, false, 5339744442951932512708.00, "", false, false, false, "4P", 15, "ZT3L", "1970-01-01", true, true, 8723978004792.841427);
INSERT INTO t1 VALUES (NULL, "UE", "", "1970-01-01 08:00:00", NULL, 4918746116215082509251.58, "F4", NULL, false, false, "", 6, NULL, "1970-01-01", false, true, 9679273849248.785024);
INSERT INTO t1 VALUES (28671194.3123163209282, NULL, "yj", NULL, true, 6973845273123480126325.85, NULL, true, false, true, "I5r", 12, "g", "1970-01-01", true, false, NULL);
INSERT INTO t0 VALUES (NULL);
INSERT INTO t0 VALUES (true), (false);
INSERT INTO t0 VALUES (true);
INSERT INTO t1 VALUES (43942442.3528499881008, "8", "HSj", "1970-01-01 08:00:00", false, 9194558332252391608108.15, NULL, NULL, false, NULL, "", 7, "c", "1970-01-01", true, true, 7013944740911.131518);
INSERT INTO t1 VALUES (6463033.1217758883685, "9Oo", "", "1970-01-01 08:00:00", false, 1205362363109081513812.37, "AQ", true, false, true, "", 6, "", "1970-01-01", false, false, NULL);
INSERT INTO t0 VALUES (true), (false);
INSERT INTO t1 VALUES (10241252.5559037922491, "", "x", "1970-01-01 08:00:00", true, 2734397181379968344591.44, NULL, false, NULL, false, "A", 1, "vmFB", "1970-01-01", false, true, 3121021021148.610493), (61187659.6283336945797, "", "", "1970-01-01 08:00:00", false, 7866283758529597491815.37, "7", true, true, false, "", 12, "nNSGJCAz", "1970-01-01", true, false, 2296836055051.780341);
INSERT INTO t1 VALUES (NULL, NULL, "DX", "1970-01-01 08:00:00", false, 5261485186743293241441.87, "J", false, false, false, "e", 12, "v", "1970-01-01", false, true, 5504419294039.582228), (NULL, NULL, "a06Oy", "1970-01-01 08:00:00", false, 3232882271815249453464.96, "6", false, true, true, "9C", 15, "f", "1970-01-01", true, true, 1041960297318.255118), (39839715.7946509792395, "6a", "", "1970-01-01 08:00:00", true, 597531077617894368360.56, "", true, true, false, "5gJYLAu", 6, "X", "1970-01-01", false, false, 5039085744210.151314);
INSERT INTO t1 VALUES (63160206.5417195472350, "ySOo", "leTNIqYJ", "1970-01-01 08:00:00", true, 5032514156737484256638.67, "", true, false, false, "M", 0, "bl", "1970-01-01", NULL, false, 9161177495328.516695), (NULL, NULL, "P", "1970-01-01 08:00:00", true, 3650744619408889946977.43, "", false, false, true, "cuxp", 7, "", "1970-01-01", true, false, NULL), (NULL, "t", "", "1970-01-01 08:00:00", true, 3424900285977550140702.40, NULL, true, NULL, true, "hVUG", 0, NULL, "1970-01-01", false, true, 1923126631481.155041);
INSERT INTO t1 VALUES (90629722.9650371678969, "nBEJM", NULL, "1970-01-01 08:00:00", true, NULL, "rE", true, NULL, true, "qY", 7, "9", "1970-01-01", NULL, false, 2581824087881.613232), (66965631.7748945233585, "viE5", "HJSoM", "1970-01-01 08:00:00", true, 9464658574082811526310.83, "tp", false, true, false, "v", 8, "gY", NULL, true, true, 2700012311029.479748);
INSERT INTO t0 VALUES (false), (true);
INSERT INTO t1 VALUES (56008714.0265645172472, "", "", NULL, NULL, 9776559882202242591074.09, "", true, false, true, "", 12, "V", NULL, true, true, NULL);
INSERT INTO t0 VALUES (NULL), (true);
INSERT INTO t0 VALUES (false), (true), (false);
2.mysql> SELECT COUNT(*) FROM t1 WHERE NULLIF(CAST(CAST(1.987956969E9 AS STRING )  AS BOOLEAN ) , CASE  WHEN  ( t1.c_1_7 ) != ( t1.c_1_7 )  THEN  ( (("")||(NULL)) ) = ( t1.c_1_1 )  ELSE CAST(t1.c_1_5 AS BOOLEAN )  END) ;
+----------+
| count(*) |
+----------+
|        3 |
+----------+
3. mysql> SELECT SUM(count) FROM (SELECT CAST((CAST(NULLIF(CAST(CAST(1.987956969E9 AS STRING )  AS BOOLEAN ) , CASE  WHEN  ( t1.c_1_7 ) != ( t1.c_1_7 )  THEN  ( (("")||(NULL)) ) = ( t1.c_1_1 )  ELSE CAST(t1.c_1_5 AS BOOLEAN )  END)  AS BOOLEAN) = true)AS INT) as count FROM t1 ) t;
+--------------+
| sum(`count`) |
+--------------+
|            2 |
+--------------+

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - branch-2.1 Release
 - +-----------------------------+
| current_version()           |
+-----------------------------+
| SQLANCER_01_RELEASE 5bfc8b3 |
+-----------------------------+

**Comment by Seaven** [comment]
```
mysql> select NULLIF(CAST(CAST(1.987956969E9 AS STRING ) AS BOOLEAN ) , CASE WHEN ( t1.c_1_7 ) != ( t1.c_1_7 ) THEN ( (("")||(NULL)) ) = ( t1.c_1_1 ) ELSE CAST(t1.c_1_5 AS BOOLEAN ) END) as x , * from t1;
+------+------------------------+-------+----------+---------------------+-------+---------------------------+-------+-------+-------+-------+---------+--------+----------+------------+--------+--------+----------------------+
| x    | c_1_0                  | c_1_1 | c_1_2    | c_1_3               | c_1_4 | c_1_5                     | c_1_6 | c_1_7 | c_1_8 | c_1_9 | c_1_10  | c_1_11 | c_1_12   | c_1_13     | c_1_14 | c_1_15 | c_1_16               |
+------+------------------------+-------+----------+---------------------+-------+---------------------------+-------+-------+-------+-------+---------+--------+----------+------------+--------+--------+----------------------+
| NULL |                   NULL | t     |          | 1970-01-01 08:00:00 |     1 | 3424900285977550140702.40 | NULL  |     1 |  NULL |     1 | hVUG    |      0 | NULL     | 1970-01-01 |      0 |      1 | 1923126631481.155041 |
| NULL |  2087436.8227650056287 | oLN   | oWCnikc  | 1970-01-01 08:00:00 |     0 | 6013717296855354277422.32 | NULL  |     0 |     1 |  NULL | EJDV8M  |      7 | L632W    | 1970-01-01 |      0 |      0 | 3486248642558.852308 |
| NULL | 63160206.5417195472350 | ySOo  | leTNIqYJ | 1970-01-01 08:00:00 |     1 | 5032514156737484256638.67 |       |     1 |     0 |     0 | M       |      0 | bl       | 1970-01-01 |   NULL |      0 | 9161177495328.516695 |
|    1 | 90629722.9650371678969 | nBEJM | NULL     | 1970-01-01 08:00:00 |     1 |                      NULL | rE    |     1 |  NULL |     1 | qY      |      7 | 9        | 1970-01-01 |   NULL |      0 | 2581824087881.613232 |
| NULL | 28671194.3123163209282 | NULL  | yj       | NULL                |     1 | 6973845273123480126325.85 | NULL  |     1 |     0 |     1 | I5r     |     12 | g        | 1970-01-01 |      1 |      0 |                 NULL |
|  168 | 61187659.6283336945797 |       |          | 1970-01-01 08:00:00 |     0 | 7866283758529597491815.37 | 7     |     1 |     1 |     0 |         |     12 | nNSGJCAz | 1970-01-01 |      1 |      0 | 2296836055051.780341 |
| NULL | 10241252.5559037922491 |       | x        | 1970-01-01 08:00:00 |     1 | 2734397181379968344591.44 | NULL  |     0 |  NULL |     0 | A       |      1 | vmFB     | 1970-01-01 |      0 |      1 | 3121021021148.610493 |
| NULL |  6463033.1217758883685 | 9Oo   |          | 1970-01-01 08:00:00 |     0 | 1205362363109081513812.37 | AQ    |     1 |     0 |     1 |         |      6 |          | 1970-01-01 |      0 |      0 |                 NULL |
| NULL | 15657298.2318305431585 |       |          | NULL                |     0 |    5339744442951932512708 |       |     0 |     0 |     0 | 4P      |     15 | ZT3L     | 1970-01-01 |      1 |      1 | 8723978004792.841427 |
| NULL | 39839715.7946509792395 | 6a    |          | 1970-01-01 08:00:00 |     1 |  597531077617894368360.56 |       |     1 |     1 |     0 | 5gJYLAu |      6 | X        | 1970-01-01 |      0 |      0 | 5039085744210.151314 |
|    1 |                   NULL | 70    | l        | 1970-01-01 08:00:00 |     0 |                      NULL |       |     0 |     0 |     0 | 4SQ22D  |     15 | l        | 1970-01-01 |      0 |      1 | 7019425131308.978094 |
|   45 | 56008714.0265645172472 |       |          | NULL                |  NULL | 9776559882202242591074.09 |       |     1 |     0 |     1 |         |     12 | V        | NULL       |      1 |      1 |                 NULL |
| NULL |                   NULL | NULL  | P        | 1970-01-01 08:00:00 |     1 | 3650744619408889946977.43 |       |     0 |     0 |     1 | cuxp    |      7 |          | 1970-01-01 |      1 |      0 |                 NULL |
| NULL |                   NULL | NULL  | a06Oy    | 1970-01-01 08:00:00 |     0 | 3232882271815249453464.96 | 6     |     0 |     1 |     1 | 9C      |     15 | f        | 1970-01-01 |      1 |      1 | 1041960297318.255118 |
| NULL | 29392878.4859718057343 | Q     | U        | 1970-01-01 08:00:00 |     0 | 4330076166445984142754.25 | kVuD  |     1 |     0 |  NULL | 6m      |      2 | 8pncW    | 1970-01-01 |      1 |      0 | 9034030916002.091349 |
| NULL | 43942442.3528499881008 | 8     | HSj      | 1970-01-01 08:00:00 |     0 | 9194558332252391608108.15 | NULL  |  NULL |     0 |  NULL |         |      7 | c        | 1970-01-01 |      1 |      1 | 7013944740911.131518 |
| NULL | 66965631.7748945233585 | viE5  | HJSoM    | 1970-01-01 08:00:00 |     1 | 9464658574082811526310.83 | tp    |     0 |     1 |     0 | v       |      8 | gY       | NULL       |      1 |      1 | 2700012311029.479748 |
| NULL | 73274266.2559530799767 | x     | y        | 1970-01-01 08:00:00 |     1 | 9795931244396562931345.34 |       |     1 |     1 |     1 | Vty     |      0 | yyf      | 1970-01-01 |   NULL |      0 | 7148382338283.990529 |
| NULL |                   NULL | NULL  | DX       | 1970-01-01 08:00:00 |     0 | 5261485186743293241441.87 | J     |     0 |     0 |     0 | e       |     12 | v        | 1970-01-01 |      0 |      1 | 5504419294039.582228 |
| NULL |                   NULL | UE    |          | 1970-01-01 08:00:00 |  NULL | 4918746116215082509251.58 | F4    |  NULL |     0 |     0 |         |      6 | NULL     | 1970-01-01 |      0 |      1 | 9679273849248.785024 |
+------+------------------------+-------+----------+---------------------+-------+---------------------------+-------+-------+-------+-------+---------+--------+----------+------------+--------+--------+----------------------+
20 rows in set (0.01 sec)

mysql> select NULLIF(CAST(CAST(1.987956969E9 AS STRING ) AS BOOLEAN ) , CASE WHEN ( t1.c_1_7 ) != ( t1.c_1_7 ) THEN ( (("")||(NULL)) ) = ( t1.c_1_1 ) ELSE CAST(t1.c_1_5 AS BOOLEAN ) END) as x , * from t1 order by x;
+------+------------------------+-------+-------+---------------------+-------+---------------------------+-------+-------+-------+-------+--------+--------+--------+------------+--------+--------+----------------------+
| x    | c_1_0                  | c_1_1 | c_1_2 | c_1_3               | c_1_4 | c_1_5                     | c_1_6 | c_1_7 | c_1_8 | c_1_9 | c_1_10 | c_1_11 | c_1_12 | c_1_13     | c_1_14 | c_1_15 | c_1_16               |
+------+------------------------+-------+-------+---------------------+-------+---------------------------+-------+-------+-------+-------+--------+--------+--------+------------+--------+--------+----------------------+
| NULL |                   NULL | t     |       | 1970-01-01 08:00:00 |     1 | 3424900285977550140702.40 | NULL  |     1 |  NULL |     1 | hVUG   |      0 | NULL   | 1970-01-01 |      0 |      1 | 1923126631481.155041 |
| NULL | 28671194.3123163209282 | NULL  | yj    | NULL                |     1 | 6973845273123480126325.85 | NULL  |     1 |     0 |     1 | I5r    |     12 | g      | 1970-01-01 |      1 |      0 |                 NULL |
| NULL | 10241252.5559037922491 |       | x     | 1970-01-01 08:00:00 |     1 | 2734397181379968344591.44 | NULL  |     0 |  NULL |     0 | A      |      1 | vmFB   | 1970-01-01 |      0 |      1 | 3121021021148.610493 |
| NULL |  6463033.1217758883685 | 9Oo   |       | 1970-01-01 08:00:00 |     0 | 1205362363109081513812.37 | AQ    |     1 |     0 |     1 |        |      6 |        | 1970-01-01 |      0 |      0 |                 NULL |
| NULL |                   NULL | NULL  | P     | 1970-01-01 08:00:00 |     1 | 3650744619408889946977.43 |       |     0 |     0 |     1 | cuxp   |      7 |        | 1970-01-01 |      1 |      0 |                 NULL |
| NULL | 66965631.7748945233585 | viE5  | HJSoM | 1970-01-01 08:00:00 |     1 | 9464658574082811526310.83 | tp    |     0 |     1 |     0 | v      |      8 | gY     | NULL       |      1 |      1 | 2700012311029.479748 |
| NULL |                   NULL | NULL  | DX    | 1970-01-01 08:00:00 |     0 | 5261485186743293241441.87 | J     |     0 |     0 |     0 | e      |     12 | v      | 1970-01-01 |      0 |      1 | 5504419294039.582228 |
|    1 | 90629722.9650371678969 | nBEJM | NULL  | 1970-01-01 08:00:00 |     1 |                      NULL | rE    |     1 |  NULL |     1 | qY     |      7 | 9      | 1970-01-01 |   NULL |      0 | 2581824087881.613232 |
|    1 |                   NULL | 70    | l     | 1970-01-01 08:00:00 |     0 |                      NULL |       |     0 |     0 |     0 | 4SQ22D |     15 | l      | 1970-01-01 |      0 |      1 | 7019425131308.978094 |
+------+------------------------+-------+-------+---------------------+-------+---------------------------+-------+-------+-------+-------+--------+--------+--------+------------+--------+--------+----------------------+
9 rows in set (0.01 sec)

mysql> select NULLIF(CAST(CAST(1.987956969E9 AS STRING ) AS BOOLEAN ) , CASE WHEN ( t1.c_1_7 ) != ( t1.c_1_7 ) THEN ( (("")||(NULL)) ) = ( t1.c_1_1 ) ELSE CAST(t1.c_1_5 AS BOOLEAN ) END) as x , * from t1 order by x;
```

**Comment by Seaven** [comment]
I found other bug on branch-2.1

**Comment by Seaven** [comment]
And the result error bug was fixd in master BE.

**Comment by stdpain** [comment]
mini reproduce case:
```
select NULLIF(CAST(CAST(1.987956969E9 AS STRING ) AS BOOLEAN ) , CASE WHEN ( t1.c_1_7 ) != ( t1.c_1_7 ) THEN ( (("")||(NULL)) ) = ( t1.c_1_1 ) ELSE CAST(t1.c_1_5 AS BOOLEAN ) END) as x,t1.c_1_7,t1.c_1_5 from t1;
```
```
    @          0x8d3999d  google::LogMessage::Fail()
    @          0x8d3bc79  google::LogMessage::SendToLog()
    @          0x8d39519  google::LogMessage::Flush()
    @          0x8d3c279  google::LogMessageFatal::~LogMessageFatal()
    @          0x5a59c9e  starrocks::vectorized::Chunk::check_or_die()
    @          0x5a539c2  starrocks::vectorized::Chunk::append_column()
    @          0x65a7528  starrocks::vectorized::ProjectNode::get_next()
    @          0x4f7a2e3  starrocks::PlanFragmentExecutor::_get_next_internal_vectorized()
    @          0x4f78646  starrocks::PlanFragmentExecutor::_open_internal_vectorized()
    @          0x4f7786e  starrocks::PlanFragmentExecutor::open()
    @          0x4e0097a  starrocks::FragmentExecState::execute()
    @          0x4e05891  starrocks::FragmentMgr::exec_actual()
    @          0x4e06297  _ZZN9starrocks11FragmentMgr18exec_plan_fragmentERKNS_23TExecPlanFragmentParamsERKSt8functionIFvPNS_20PlanFragmentExecutorEEESA_ENKUlvE_clEv
    @          0x4e0ef48  _ZNSt17_Function_handlerIFvvEZN9starrocks11FragmentMgr18exec_plan_fragmentERKNS1_23TExecPlanFragmentParamsERKSt8functionIFvPNS1_20PlanFragmentExecutorEEESC_EUlvE_E9_M_invokeERKSt9_Any_data
    @          0x3f200a0  std::function<>::operator()()
    @          0x52150f0  starrocks::FunctionRunnable::run()
    @          0x5211a53  starrocks::ThreadPool::dispatch_thread()
    @          0x522cf1d  std::__invoke_impl<>()
    @          0x522bd61  std::__invoke<>()
    @          0x5229c6e  _ZNSt5_BindIFMN9starrocks10ThreadPoolEFvvEPS1_EE6__callIvJEJLm0EEEET_OSt5tupleIJDpT0_EESt12_Index_tupleIJXspT1_EEE
    @          0x5227742  std::_Bind<>::operator()<>()
    @          0x522287e  std::_Function_handler<>::_M_invoke()
    @          0x3f200a0  std::function<>::operator()()
    @          0x52421b3  starrocks::Thread::supervise_thread()
    @     0x7f095ff7eea5  start_thread
    @     0x7f095f383b0d  __clone
    @              (nil)  (unknown)
```

**Comment by stdpain** [comment]
Fixed by #3022
